pkg/time/multiple.go

```diff
@@ -0,0 +1,109 @@
+package time
+
+import (
+	"errors"
+	"sync"
+	"time"
+)
+
+// Ceil returns the result of rounding t up to a multiple of d (since the zero time).
```

Comment:
Based on time module documentation, it looks like Truncate rounds down and Round is what rounds up; not sure the behavior you want:
https://golang.org/pkg/time/#Time.Truncate

Reply:
`time.Round()` rounds up or down to the nearest. I wanted a func that always goes up.

Reply:
Nevermind, I see now that I completely neglected the Add() part of this function.

Reply:
I actually want to double-down on my initial comment here; I agree that Ceil should always round up, but you get behavior like this when you supply certain values:

```
func main() {
	now := time.Now()
	d, _ := time.ParseDuration("50m")
	fmt.Println(d)
	fmt.Println(now)
	fmt.Println(now.Add(d))
	fmt.Println(now.Add(d).Truncate(d))
}

```
Returns this:
```
50m0s
2009-11-10 23:00:00 +0000 UTC m=+0.000000001
2009-11-10 23:50:00 +0000 UTC m=+3000.000000001
2009-11-10 23:20:00 +0000 UTC
```
The Truncate forces the final time to be earlier than the Added time (because 23:20 is 1400 minutes, the rounded-down multiple of 50).  Is this expected behavior?  If so we should change the name from Ceil to something else -- or if we only want to accept multiples of 10 or discrete units as Duration arguments, we should build that into the function.  

We should also probably write some tests for this function as well.  I'd be happy to try my hand at it once we've established whether the function is behaving as expected/needed.

Reply:
That is the expected behaviour, yes, although not because 23:20 is 1400 minutes. `time.Time.Truncate()`, `time.Time.Round()`, and `Ceil()` all operate relative to the Zero Time (January 1, year 1, 00:00:00.000000000 UTC), so `2009-11-10 23:20:00 UTC` is [some multiple](https://www.wolframalpha.com/input/?i=(0001-1-1T00:00:00+UTC+to+2009-11-10T23:20:00+UTC)+%2F+50+minutes) of 50 minutes from that time. Check out the [ruminations in the `time` src about computations on time](https://golang.org/src/time/time.go#L346).

I spent two days thinking about how to do semantically-nice time multiples for the snapshots before giving up. The most important behaviour of `Ceil()` is that it's deterministic i.e. the snapshooter will always choose the same snapshot times no matter when it was started.